Feature: auto scale batch size (#1638)
PyTorchLightning/pytorch-lightning: docs/source/training_tricks.rst

    # clip gradients with norm above 0.5
    trainer = Trainer(gradient_clip_val=0.5)

> Auto scaling of batch size
> -------------------------------------

Reviewer: ```suggestion
Auto scaling of batch size
--------------------------
```

Auto scaling of batch size may be enabled to find the largest batch size that fits into

> memory. Larger batch size often give better estimates of gradients, but may also give
> longer training time.

Reviewer 2: ```suggestion
memory. Larger batch size often yields better estimates of gradients, but may also result in
longer training time.
```

.. seealso:: :class:`~pytorch_lightning.trainer.trainer.Trainer`

.. code-block:: python

    # DEFAULT (ie: don't scale batch size automatically)
    trainer = Trainer(auto_scale_batch_size=False)

    # Autoscale batch size 
    trainer = Trainer(auto_scale_batch_size=True|'power'|'binsearch')

> Setting the feature to `True` enables `'power'` scaling, that starting from a

Reviewer: so `True` == `'power'` ?

Author: yes

> batch size of 1 keeps double the batch size until an out-of-memory (OMM) error is
> encountered. Setting the argument to `'binsearch'` continue to finetune the batch
> size by duing a binary search. 

Reviewer 2: ```suggestion
batch size of 1 keeps doubling the batch size until an out-of-memory (OOM) error is
encountered. Setting the argument to `'binsearch'` continues to finetune the batch
size by performing a binary search. 
```

.. note:: 

>     This feature expects that a `batch_size` field exist in the `hparams` of your model i.e.
>     `model.hparams.batch_size` should exist and will be overridden by the results of this
>     algorithm.

Reviewer 2: ```suggestion
    This feature expects that a `batch_size` field in the `hparams` of your model, i.e.,
    `model.hparams.batch_size` should exist and will be overridden by the results of this
    algorithm.
```

> The scaling algorithm has a number of parameters, that the user can control by
> invoking the trainer method `.scale_batch_size` themself.

Reviewer 2: ```suggestion
The scaling algorithm has a number of parameters that the user can control by
invoking the trainer method `.scale_batch_size` themself.
```
